src/wallet/sqlite.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_WALLET_SQLITE_H
+#define BITCOIN_WALLET_SQLITE_H
+
+#include <wallet/db.h>
+
+struct bilingual_str;
+class SQLiteDatabase;
+
+/** RAII class that provides access to a WalletDatabase */
+class SQLiteBatch : public DatabaseBatch
+{
+private:
+    SQLiteDatabase& m_database;
+
+    bool m_read_only = false;
+
+    bool ReadKey(CDataStream&& key, CDataStream& value) override;
+    bool WriteKey(CDataStream&& key, CDataStream&& value, bool overwrite=true) override;
```

Comment:
1a2eec03f0dc7731d5c199a97db148534ce174a2, nit -- style:
```suggestion
    bool WriteKey(CDataStream&& key, CDataStream&& value, bool overwrite = true) override;
```

Reply:
Done